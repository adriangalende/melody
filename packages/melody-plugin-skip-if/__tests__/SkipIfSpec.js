import { compile, toString } from 'melody-compiler';
import { extension as coreExtension } from 'melody-extension-core';
import idomPlugin from 'melody-plugin-idom';
import skipIfPlugin from '../src';

test('skip if client should skip content', () => {
    expect(
        render(`<div>{% skip if client %}hello world{% endskip %}</div>`)
    ).toMatchSnapshot();
});

test('skip if server should be ignored', () => {
    expect(
        render(`<div>{% skip if server %}hello world{% endskip %}</div>`)
    ).toMatchSnapshot();
});

test('skip if defined should check if parent element is defined', () => {
    expect(
        render(`<x-div>{% skip if defined %}hello world{% endskip %}</x-div>`)
    ).toMatchSnapshot();
});

test('skip if defined should fail if parent element is not a custom element', () => {
    expect(
        render(`<div>{% skip if defined %}hello world{% endskip %}</div>`)
    ).toMatchSnapshot();
});

function render(template) {
    try {
        const jsTemplate = compile(
            'test.twig',
            template,
            coreExtension,
            idomPlugin,
            skipIfPlugin
        );
        return toString(jsTemplate, template).code;
    } catch (e) {
        return e.message;
    }
}
