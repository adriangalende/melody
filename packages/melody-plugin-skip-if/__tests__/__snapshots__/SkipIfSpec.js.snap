// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skip if client should skip content 1`] = `
"import { skip, elementOpen, elementClose } from \\"melody-idom\\";
export const _template = {};

_template.render = function (_context) {
  elementOpen(\\"div\\", null, null);
  skip();
  elementClose(\\"div\\");
};

if (process.env.NODE_ENV !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

export default function Test(props) {
  return _template.render(props);
}"
`;

exports[`skip if defined should check if parent element is defined 1`] = `
"import { text, skip, elementOpen, elementClose } from \\"melody-idom\\";
export const _template = {};

_template.render = function (_context) {
  elementOpen(\\"x-div\\", null, null);

  if (customElements.get(\\"x-div\\") !== undefined) {
    skip();
  } else {
    text(\\"hello world\\");
  }

  elementClose(\\"x-div\\");
};

if (process.env.NODE_ENV !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

export default function Test(props) {
  return _template.render(props);
}"
`;

exports[`skip if defined should fail if parent element is not a custom element 1`] = `
"skip if defined can only be used inside of a custom element
> 1 | <div>{% skip if defined %}hello world{% endskip %}</div>
    |  ^^^

Custom Element must contain a \\"-\\" within their name but \\"div\\" does not seem to contain one.
More information about custom elements can be found here: https://developer.mozilla.org/en-US/docs/Web/Web_Components/Custom_Elements"
`;

exports[`skip if server should be ignored 1`] = `
"import { text, elementOpen, elementClose } from \\"melody-idom\\";
export const _template = {};

_template.render = function (_context) {
  elementOpen(\\"div\\", null, null);
  text(\\"hello world\\");
  elementClose(\\"div\\");
};

if (process.env.NODE_ENV !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

export default function Test(props) {
  return _template.render(props);
}"
`;
